index.bs

```diff
@@ -5483,6 +5500,415 @@ At this time, one [=credential property=] is defined: the [=resident key credent
+ 1. Let |credentialId| be the result of decoding [=Base64url Encoding=] on the |parameters|' |credentialId| property.
+ 1. If |credentialId| is failure, return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Let |isResidentCredential| be the |parameters|' |isResidentCredential| property.
+ 1. If |isResidentCredential| is not defined, return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Let |rpId| be the |parameters|' |rpId| property.
+ 1. If |rpId| is not a valid [=RP ID=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Let |privateKey| be the result of decoding [=Base64url Encoding=] on the |parameters|' |privateKey| property.
+ 1. If |privateKey| is failure, return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. If |privateKey| is not a validly-encoded asymmetric key package containing a single ECDSA private key on the P-256
+     curve per [[RFC5958]], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. If the |parameters|' |userHandle| property is defined:
+     1. Let |userHandle| be the result of decoding [=Base64url Encoding=] on the |parameters|' |userHandle| property.
+     1. If |userHandle| is failure, return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Otherwise:
+    1. If |isResidentCredential| is [TRUE], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+    1. Let |userHandle| be `null`.
+ 1. If |authenticatorId| does not match any [=Virtual Authenticator=] stored in the [=Virtual Authenticator
+     Database=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Let |authenticator| be the [=Virtual Authenticator=] matched by |authenticatorId|.
+ 1. If |isResidentCredential| is [TRUE] and the |authenticator|'s |hasResidentKey| property is [FALSE], return a
+     [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Let |credential| be a new [=Client-side-resident Public Key Credential Source=] if |isResidentCredential| is [TRUE]
+     or a [=Server-side-resident Public Key Credential Source=] otherwise whose items are:
+    : [=public key credential source/type=]
+    :: {{public-key}}
+    : [=public key credential source/id=]
+    :: |credentialId|
+    : [=public key credential source/privateKey=]
+    :: |privateKey|
+    : [=public key credential source/rpId=]
+    :: |rpId|
+    : [=public key credential source/userHandle=]
+    :: |userHandle|
+ 1. Associate a [=signature counter=] |counter| to the |credential| with a starting value equal to the |parameters|'
+     |signCount| or `0` if |signCount| is `null`.
+ 1. Store the |credential| and |counter| in the database of the |authenticator|.
+ 1. Return [=success=].
+
+## <dfn>Get Credentials</dfn> ## {#sctn-automation-get-credentials}
```

Comment:
Is there a use case for this? The webdriver philosophy is to only add a `getter` type command if there is a sufficient need and that tests aren't required to be aware of this info in the first place. This feels like you might be setting it once per test and might never call this API but happy to be corrected

Reply:
A getter for the credentials gives a client of the API more granularity. For example, a relying party might want to run a test where they create a credential via `navigator.credentials.create` and then produce an assertion with `navigator.credentials.get`. Being able to inspect the credential stored in the virtual authenticator by `navigator.credentials.create` lets them verify the creation ran as expected, so if the assertion fails the problem is with the `get` call.

A big motivation for this API is to allow RPs to test their webauthn client code, as well as allow end-to-end tests. We should probably avoid being too restrictive on its design.

Reply:
ok.